ray-operator/controllers/ray/raycluster_controller.go

```diff
@@ -1229,19 +1229,19 @@ func (r *RayClusterReconciler) calculateStatus(ctx context.Context, instance *ra
 }
 
 // Best effort to obtain the ip of the head node.
-func (r *RayClusterReconciler) getHeadPodIP(ctx context.Context, instance *rayv1.RayCluster) (string, error) {
+func (r *RayClusterReconciler) getHeadPodIPAndName(ctx context.Context, instance *rayv1.RayCluster) (string, string, error) {
 	logger := ctrl.LoggerFrom(ctx)
 
 	runtimePods := corev1.PodList{}
```

Comment:
Can we use [RayClusterHeadPodsAssociationOptions](https://github.com/ray-project/kuberay/blob/ca39dc94f93592efc0ee7093b4fcded2a01cb7e5/ray-operator/controllers/ray/common/association.go#L78) instead?